src/CoaddPsf.cc

```diff
@@ -354,8 +354,11 @@ class CoaddPsf::Factory : public tbl::io::PersistableFactory {
 
     virtual PTR(tbl::io::Persistable)
     read(InputArchive const & archive, CatalogVector const & catalogs) const {
-        CoaddPsfPersistenceHelper const & keys1 = CoaddPsfPersistenceHelper::get();
+        if (catalogs.size() == 1u) {
```

Comment:
Please explain this test (why does catalogs.size() tell us the catalog version) in a code comment.